pkg/cache/types/types.go

```diff
@@ -9,6 +11,13 @@ type Resource interface {
 	proto.Message
 }
 
+// ResourceWithTtl is a Resource with an optional TTL.
+type ResourceWithTtl struct {
+	Resource Resource
+
+	Ttl *time.Duration
```

Comment:
is the Ttl config set in DiscoveryResponse? How will the control plane know what is a good  value to set for heartbeat interval?

Reply:
Both TTL and heartbeat interval is specified by the user, so it's up to them to decide